src/Button/IconButton.tsx

```diff
@@ -4,20 +4,62 @@ import {ForwardRefComponent as PolymorphicForwardRefComponent} from '../utils/po
 import {ButtonBase} from './ButtonBase'
 import {defaultSxProp} from '../utils/defaultSxProp'
 import {generateCustomSxProp} from './Button'
+import {Tooltip} from '../drafts/Tooltip/Tooltip'
 
-const IconButton = forwardRef(({sx: sxProp = defaultSxProp, icon: Icon, ...props}, forwardedRef): JSX.Element => {
-  let sxStyles = sxProp
-  // grap the button props that have associated data attributes in the styles
-  const {size} = props
+const IconButton = forwardRef(
+  (
+    {
+      sx: sxProp = defaultSxProp,
+      icon: Icon,
+      'aria-label': ariaLabel,
+      description,
+      disabled,
+      // This is a temporary prop to incrementally introduce tooltips to the IconButton component.
+      _enableTooltip_ = false,
+      ...props
+    },
+    forwardedRef,
+  ): JSX.Element => {
+    let sxStyles = sxProp
+    // grap the button props that have associated data attributes in the styles
+    const {size} = props
 
-  if (sxProp !== null && Object.keys(sxProp).length > 0) {
-    sxStyles = generateCustomSxProp({size}, sxProp)
-  }
+    if (sxProp !== null && Object.keys(sxProp).length > 0) {
+      sxStyles = generateCustomSxProp({size}, sxProp)
+    }
 
-  return (
-    // @ts-expect-error StyledButton wants both Anchor and Button refs
-    <ButtonBase icon={Icon} data-component="IconButton" sx={sxStyles} type="button" {...props} ref={forwardedRef} />
-  )
-}) as PolymorphicForwardRefComponent<'button' | 'a', IconButtonProps>
+    const displayTooltip = _enableTooltip_ && !disabled && ariaLabel
```

Comment:
Since we have the ability to keep the tooltip off on icon buttons, I am thinking not to handle the case where icon buttons that are already wrapped with Tooltip, to keep the implementation lean. 


```
<Tooltip text="Notifications" type="label" >
  <IconButton icon={X} aria-label="Notifications" />
</Tooltip>
```

Because this usage, is [discourage by the Tooltip accessibility checks](https://github.com/primer/react/blob/main/src/drafts/Tooltip/Tooltip.tsx#L228) as well as have duplicated info (text and aria-label). I am thinking we shouldn't cover this case in the implementation because it is not a usage we encourage. I would love to hear your thoughts on this 🙌🏻